python/GafferVDBTest/SphereLevelSetTest.py

```diff
@@ -0,0 +1,117 @@
+#        written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+#  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+#  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+#  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+#  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+#  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+#  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+#  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+#  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+#  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+#  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+##########################################################################
+
+import GafferVDB
+import GafferVDBTest
+
+import imath
+
+class SphereLevelSetTest( GafferVDBTest.VDBTestCase ) :
+
+    def testCanCreateSphereLevelSet( self ) :
+        sphereLevelSet = GafferVDB.SphereLevelSet( "SphereLevelSet" )
+        surfaceGrid = sphereLevelSet['out'].object( "vdb" ).findGrid( "surface" )
+        self.assertTrue( surfaceGrid is not None )
+        self.assertEqual( surfaceGrid.gridClass, "level set" )
+
+    def testCanSetGridName( self ) :
+        sphereLevelSet = GafferVDB.SphereLevelSet( "SphereLevelSet" )
+        sphereLevelSet["grid"].setValue( "foo" )
+
+        surfaceGrid = sphereLevelSet['out'].object( "vdb" ).findGrid( "foo" )
+        self.assertTrue( surfaceGrid is not None )
+
+    def testSignedDistanceIsNearlyZeroForPointsOnSphere( self ) :
+        sphereLevelSet = GafferVDB.SphereLevelSet( "SphereLevelSet" )
+        surfaceGrid = sphereLevelSet['out'].object( "vdb" ).findGrid( "surface" )
+        accessor = surfaceGrid.getAccessor()
+        # default voxel size is 0.1 and radius 1.0 so the radius of the sphere in index space is 10
+        self.assertEqual( accessor.probeValue( ( 10, 0, 0 ) ) , ( 0.0, True ) )
+        self.assertEqual( accessor.probeValue( ( 0, 10, 0 ) ) , ( 0.0, True ) )
+        self.assertEqual( accessor.probeValue( ( 0, 0, 10 ) ) , ( 0.0, True ) )
+
+    def testCanSetSphereRadius( self ) :
+        sphereLevelSet = GafferVDB.SphereLevelSet( "SphereLevelSet" )
+        sphereLevelSet["radius"].setValue( 2.0 )
+
+        surfaceGrid = sphereLevelSet['out'].object( "vdb" ).findGrid( "surface" )
+        accessor = surfaceGrid.getAccessor()
+        # default voxel size is 0.1 and radius 1.0 so the radius of the sphere in index space is 10
+        self.assertEqual( accessor.probeValue( ( 20, 0, 0 ) ) , ( 0.0, True ) )
+        self.assertEqual( accessor.probeValue( ( 0, 20, 0 ) ) , ( 0.0, True ) )
+        self.assertEqual( accessor.probeValue( ( 0, 0, 20 ) ) , ( 0.0, True ) )
+
+    def testBounds( self ) :
```

Comment:
Just verifying visually in the viewer, I'm seeing the voxel visualisation exceeding the bounding box slightly. I presume this problem would originate in `IECoreVDB::VDBObject` rather than in this PR though.

Reply:
Interesting. I'm not happy with the bounds tests and why I can write tests which aren't accurate to  a decent epsilon. I'll query it on the vdb list.

Reply:
Cool - no need to hold up this PR for it, but I'd be interested to hear what you find.